### PR TITLE
Remove deprecated configuration keys or APIs

### PR DESCRIPTION
**Pull request by PROFeNoM**
### Description

AIT-9171

**APIs**

- dd_trace_forward_call
- dd_trace_generate_id
- dd_trace_push_span_id
- dd_trace_pop_span_id
- additional_trace_meta
- Everything that is circuit-breaker-related
  - dd_tracer_circuit_breaker_register_error
  - dd_tracer_circuit_breaker_register_success
  - dd_tracer_circuit_breaker_can_try
  - dd_tracer_circuit_breaker_info
- Everything that may have been marked as `@deprecated` in the "Legacy PHP API"

**Config Knobs**

- DD_INTEGRATIONS_DISABLED → DD_TRACE_[INTEGRATION]_ENABLE=false
- ~DDTRACE_REQUEST_INIT_HOOK → DD_TRACE_REQUEST_INIT_HOOK~
- DD_SERVICE_NAME → DD_SERVICE
- DD_TRACE_GLOBAL_TAGS → DD_TAGS
- DD_TRACE_APP_NAME → DD_SERVICE
- ddtrace_app_name → DD_SERVICE
- DD_TRACE_RESOURCE_URI_MAPPING → DD_TRACE_RESOURCE_URI_MAPPING_INCOMING and DD_TRACE_RESOURCE_URI_MAPPING_OUTGOING
- DD_SAMPLING_RATE → DD_TRACE_SAMPLE_RATE
- DD_PROPAGATION_STYLE_INJECT → DD_TRACE_PROPAGATION_STYLE_INJECT
- DD_PROPAGATION_STYLE_EXTRACT → DD_TRACE_PROPAGATION_STYLE_EXTRACT
- Everything that is circuit-breaker-related
  - DD_TRACE_AGENT_MAX_CONSECUTIVE_FAILURES
  - DD_TRACE_CIRCUIT_BREAKER_DEFAULT_MAX_CONSECUTIVE_FAILURES
  - DD_TRACE_AGENT_ATTEMPT_RETRY_TIME_MSEC
  - DD_TRACE_CIRCUIT_BREAKER_DEFAULT_RETRY_TIME_MSEC

It also removes UPGRADE-0.10.md, Http.php (unused), and circuit breaker code.

 

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

**Comment by PROFeNoM** [comment]
Note: `dd-gitlab/benchmarks-profiler` is failing and it's normal - Some changes on the profiler were made on the bp-side, but are yet to be committed to dd-trace-php

File: ext/ddtrace.c [hunk 1]
```diff
@@ -141,10 +140,10 @@ static ZEND_INI_MH(dd_OnUpdateDisabled) {
 }
 
 PHP_INI_BEGIN()
-    ZEND_INI_ENTRY("ddtrace.disable", "0", PHP_INI_SYSTEM, dd_OnUpdateDisabled)
+    ZEND_INI_ENTRY("datadog.trace.disable", "0", PHP_INI_SYSTEM, dd_OnUpdateDisabled)
```
Comment (bwoebi) [inline ext/ddtrace.c:143]: It makes sense, but this is not just dropping the deprecation and should be loudly noted with the 1.0 changes.

Comment (PROFeNoM) [reply ext/ddtrace.c:143]: Oops I committed this without wanting 😬 I was starting to do [AIT-2131](https://datadoghq.atlassian.net/browse/AIT-2131) but instead moved to smthg else (system tests) and forgot not to push this 🤦 Either way, let's do it in another PR

Comment (PROFeNoM) [reply ext/ddtrace.c:143]: The same goes for `datadog.trace.cgroup_file` and the changes in startup_logging.c

**Review by bwoebi — APPROVED** [review]
Nice cleanup :-)

**Pull request merged**